pkg/apis/openebs.io/ndm/v1alpha1/blockdeviceclaim_types.go

```diff
@@ -69,11 +69,27 @@ const (
 
 // DeviceClaimDetails defines the details of the block device that should be claimed
 type DeviceClaimDetails struct {
-	DeviceFormat   string `json:"formatType,omitempty"`     //Format of the device required, eg:ext4, xfs
-	MountPoint     string `json:"mountPoint,omitempty"`     //MountPoint of the device required. Claim device from the specified mountpoint.
-	AllowPartition bool   `json:"allowPartition,omitempty"` //AllowPartition represents whether to claim a full block device or a device that is a partition
+	// BlockVolumeMode represents whether to claim a device in Block mode or Filesystem mode.
+	// These are use cases of BlockVolumeMode:
+	// 1) Not specified: VolumeMode check will not be effective
+	// 2) VolumeModeBlock: BD should not have any filesystem or mountpoint
+	// 3) VolumeModeFileSystem: BD should have a filesystem and mountpoint. If DeviceFormat is
+	//    specified then the format should match with the FSType in BD
+	BlockVolumeMode BlockDeviceVolumeMode `json:"blockVolumeMode,omitempty"`
+	DeviceFormat    string                `json:"formatType,omitempty"`     //Format of the device required, eg:ext4, xfs
+	AllowPartition  bool                  `json:"allowPartition,omitempty"` //AllowPartition represents whether to claim a full block device or a device that is a partition
```

Comment:
comments on top & leave a line space between two fields

Reply:
@AmitKumarDas updated the code